secure-api-gateway-ig-extensions/src/main/java/com/forgerock/sapi/gateway/jws/signer/CompactSerializationJwsSigner.java

```diff
@@ -0,0 +1,198 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.forgerock.sapi.gateway.jws.signer;
+
+import static org.forgerock.openig.secrets.SecretsProviderHeaplet.secretsProvider;
+
+import java.util.ArrayList;
+import java.util.Map;
+
+import org.forgerock.json.jose.builders.JwsHeaderBuilder;
+import org.forgerock.json.jose.builders.JwtBuilderFactory;
+import org.forgerock.json.jose.jws.JwsAlgorithm;
+import org.forgerock.json.jose.jws.SigningManager;
+import org.forgerock.json.jose.jws.handlers.SigningHandler;
+import org.forgerock.json.jose.jwt.JwtClaimsSet;
+import org.forgerock.openig.heap.GenericHeaplet;
+import org.forgerock.openig.heap.HeapException;
+import org.forgerock.secrets.Purpose;
+import org.forgerock.secrets.SecretsProvider;
+import org.forgerock.secrets.keys.SigningKey;
+import org.forgerock.util.promise.Promise;
+import org.forgerock.util.promise.Promises;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * An implementation of {@link JwsSigner} to sign a {@link Map} message.s<br/>
+ * Provides a Heaplet functionality to be added in the Heap as object for use by the filters and/or the IG scripts<br/>
+ * @see CompactSerializationJwsSigner.Heaplet
+ */
+public class CompactSerializationJwsSigner implements JwsSigner {
+
+    private static final Logger logger = LoggerFactory.getLogger(CompactSerializationJwsSigner.class);
+    private final SigningManager signingManager;
+    private final Purpose<SigningKey> signingKeyPurpose;
+    private final String algorithm;
+    private final String kid;
+
+    public CompactSerializationJwsSigner(
+            SecretsProvider secretsProvider,
+            String signingKeyId,
+            String kid,
+            String algorithm
+    ) {
+        this.signingManager = new SigningManager(secretsProvider);
+        this.kid = kid;
+        this.algorithm = algorithm;
+        this.signingKeyPurpose = Purpose.purpose(signingKeyId, SigningKey.class);
+    }
+
+    /**
+     * Compute the payload signature
+     * <p>
+     *   <ul>
+     *       <li>Accepts a JSON-Based structure payload as a {@link Map} structure</li>
+     *       <p>
+     *          Payload must not be null or empty.
+     *       </p>
+     *       <li>
+     *           Accepts a critical header claims in accordance with
+     *           <a href='https://datatracker.ietf.org/doc/html/rfc7515#section-4.1.11'>(RFC7515)"crit" Header Parameter</a>
+     *           as {@link Map}<br/>
+     *           <p>
+     *               Critical header claims can be null or empty, they will be ignored in these cases.
+     *           </p>
+     *       </li>
+     *       <li>
+     *           <p>
+     *               As result, produce a JWS Compact Serialization from JWS, a string composed of three parts encoded in Base64 Url Safe and separated by a dot ( . ), using
```

Comment:
This info should be in the @return documentation

Reply:
done